source/run/include/TG4PrimaryGeneratorAction.h

```diff
@@ -48,12 +56,32 @@ class TG4PrimaryGeneratorAction : public G4VUserPrimaryGeneratorAction,
 
  private:
   // methods
+
+  G4bool CheckVMCStack(TVirtualMCStack* stack) const;
+  G4bool CheckParticleDefinition(const G4ParticleDefinition* particleDefinition,
+    const TParticle* particle) const;
+  G4bool GetProperCharge(const G4ParticleDefinition* particleDefinition,
```

Comment:
BUG: Return type should be G4double; conversion to boolean results is changing all negative charges to positive !